packer/builder/azure/driver_restapi/targets/step_stop_vm.go

```diff
@@ -18,17 +23,16 @@ type StepStopVm struct {
 }
 
 func (s *StepStopVm) Run(state multistep.StateBag) multistep.StepAction {
-	reqManager := state.Get(constants.RequestManager).(*request.Manager)
+	client := state.Get(constants.RequestManager).(management.Client)
 	ui := state.Get(constants.Ui).(packer.Ui)
 
 	errorMsg := "Error Stopping Temporary Azure VM: %s"
 
 	ui.Say("Stopping Temporary Azure VM...")
 
-	requestData := reqManager.ShutdownRoles(s.TmpServiceName, s.TmpVmName)
-	err := reqManager.ExecuteSync(requestData)
-
-	if err != nil {
+	if err := retry.ExecuteAsyncOperation(client, func() (management.OperationId, error) {
```

Comment:
hmmm... but then how would we retry the request? i.e.: 
1. Shutdown->202, 
2. OperationStatus->200 with async 409 of the retriable kind, 
3. now need to resend Shutdown... but ExecuteAsyncOperation only has an opid and no way to create the request?